tests/backup/backup_share_test.go

```diff
@@ -4215,3 +4215,177 @@ var _ = Describe("{SwapShareBackup}", func() {
+				userClusterMap[user1][clusterName] = userClusterUID
+			}
+		})
+		Step(fmt.Sprintf("Take backup of applications from the user %s", user1), func() {
+			log.InfoD(fmt.Sprintf("Taking backup of applications from the user %s", user1))
+			nonAdminCtx, err := backup.GetNonAdminCtx(user1, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", user1)
+			var wg sync.WaitGroup
+			var mu sync.RWMutex
+			userBackupMap[user1] = make(map[string]string)
+			createBackup := func(backupName string, namespace string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+				for backupLocationUID, backupLocationName := range userBackupLocationMap[user1] {
+					err := CreateBackupWithValidation(nonAdminCtx, backupName, SourceClusterName, backupLocationName, backupLocationUID, appContextsToBackup, make(map[string]string), orgID, userClusterMap[user1][SourceClusterName], "", "", "", "")
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation and validation of backup [%s] of namespace (scheduled Context) [%s]", backupName, namespace))
+					break
+				}
+				mu.Lock()
+				defer mu.Unlock()
+				userBackupMap[user1][backupName] = namespace
+			}
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					wg.Add(1)
+					go createBackup(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of user backups taken are: %v", userBackupMap)
+		})
+		Step(fmt.Sprintf("Share user %s backups with user %s with ViewOnlyAccess", user1, user2), func() {
+			log.InfoD(fmt.Sprintf("Sharing user %s backups with user %s with ViewOnlyAccess", user1, user2))
+			nonAdminCtx, err := backup.GetNonAdminCtx(user1, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", user1)
+			for backupName := range userBackupMap[user1] {
+				err := ShareBackup(backupName, nil, []string{user2}, ViewOnlyAccess, nonAdminCtx)
+				log.FailOnError(err, "failed to share user %s backup %s with user %s with ViewOnlyAccess", user1, backupName, user2)
+			}
+		})
+		Step(fmt.Sprintf("Delete the owner [%s] of the backups", user1), func() {
+			log.InfoD(fmt.Sprintf("Deleting the owner [%s] of the backups", user1))
+			err := backup.DeleteUser(user1)
+			log.FailOnError(err, "failed to delete user %s", user1)
+		})
+		Step(fmt.Sprintf("Delete user %s backups from the admin", user2), func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD(fmt.Sprintf("Deleting user %s backups from the admin", user2))
+			nonAdminCtx, err := backup.GetNonAdminCtx(user2, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", user2)
+			for backupName := range userBackupMap[user1] {
```

Comment:
Add ownerID fetch here.

Reply:
Using admin ctx to fetch the backup UID as suggested by @sn-px.